mungegithub/mungers/submit-queue.go

```diff
@@ -474,40 +467,34 @@ func (sq *SubmitQueue) internalInitialize(config *github.Config, features *featu
 			NonBlockingJobNames:  &sq.NonBlockingJobNames,
 			BuildStatus:          map[string]e2e.BuildInfo{},
 			GoogleGCSBucketUtils: gcs,
-		}).Init(admin.Mux)
+		}).Init(sharedmux.Admin)
 	}
 
 	sq.lgtmTimeCache = mungerutil.NewLabelTimeCache(lgtmLabel)
 
-	if len(mungeopts.Server.Address) > 0 {
-		if len(mungeopts.Server.WWWRoot) > 0 {
-			http.Handle("/", gziphandler.GzipHandler(http.FileServer(http.Dir(mungeopts.Server.WWWRoot))))
-		}
-		http.Handle("/prometheus", promhttp.Handler())
-		http.Handle("/prs", gziphandler.GzipHandler(http.HandlerFunc(sq.servePRs)))
-		http.Handle("/history", gziphandler.GzipHandler(http.HandlerFunc(sq.serveHistory)))
-		http.Handle("/github-e2e-queue", gziphandler.GzipHandler(http.HandlerFunc(sq.serveGithubE2EStatus)))
-		http.Handle("/merge-info", gziphandler.GzipHandler(http.HandlerFunc(sq.serveMergeInfo)))
-		http.Handle("/priority-info", gziphandler.GzipHandler(http.HandlerFunc(sq.servePriorityInfo)))
-		http.Handle("/health", gziphandler.GzipHandler(http.HandlerFunc(sq.serveHealth)))
-		http.Handle("/health.svg", gziphandler.GzipHandler(http.HandlerFunc(sq.serveHealthSVG)))
-		http.Handle("/sq-stats", gziphandler.GzipHandler(http.HandlerFunc(sq.serveSQStats)))
-		http.Handle("/flakes", gziphandler.GzipHandler(http.HandlerFunc(sq.serveFlakes)))
-		http.Handle("/metadata", gziphandler.GzipHandler(http.HandlerFunc(sq.serveMetadata)))
+	if features.Server.Enabled {
+		features.Server.Handle("/prs", gziphandler.GzipHandler(http.HandlerFunc(sq.servePRs)))
+		features.Server.Handle("/history", gziphandler.GzipHandler(http.HandlerFunc(sq.serveHistory)))
```

Comment:
That would be cleaner. We could turn this into a bunch of HandleFunc calls and put the GzipHandler at the proxy layer.

Other deployments might have a different nginx setup, though.